network/src/errors.rs

```diff
@@ -12,6 +12,7 @@ pub enum Error {
     Io(IoError),
     P2P(P2PError),
     DB(DBError),
+    Custom(String),
```

Comment:
Can we give the error a name? `Custom` is generally used in a library that reserved for user-thrown errors, it makes no sense here.